core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,307 @@
+			resp, err = nodeRequest(cl, "/v2/keys/p2p")
+			if err != nil {
+				log.Fatalf("failed to get p2p keys: %s", err)
+			}
+			var p2pKeys cmd.P2PKeyPresenters
+			if err = jsonapi.Unmarshal(resp, &p2pKeys); err != nil {
+				log.Fatalf("failed to unmarshal response body: %s", err)
+			}
+			ni.PeerId = []string{p2pKeys[0].PeerID}
+
+			// get node csa config
+			resp, err = nodeRequest(cl, "/v2/keys/csa")
+			if err != nil {
+				log.Fatalf("failed to get CSA keys: %s", err)
+			}
+			var csaKeys cmd.CSAKeyPresenters
+			if err = jsonapi.Unmarshal(resp, &csaKeys); err != nil {
+				log.Fatalf("failed to unmarshal response body: %s", err)
+			}
+			csaKey.PublicKey = csaKeys[0].PubKey
+
+			ni.CSAKeys = []*CSAKeyInfo{csaKey}
+			nodes[nodeAddresses[0]] = ni
+
+			offchainPkBytes, err := hex.DecodeString(strings.TrimPrefix(ocr2Config.OffChainPublicKey, "ocr2off_evm_"))
+			if err != nil {
+				panic(fmt.Errorf("failed to decode %s: %v", ocr2Config.OffChainPublicKey, err))
+			}
+
+			offchainPkBytesFixed := [ed25519.PublicKeySize]byte{}
+			n := copy(offchainPkBytesFixed[:], offchainPkBytes)
+			if n != ed25519.PublicKeySize {
+				panic(fmt.Errorf("wrong num elements copied"))
+			}
+
+			configPkBytes, err := hex.DecodeString(strings.TrimPrefix(ocr2Config.ConfigPublicKey, "ocr2cfg_evm_"))
+			if err != nil {
+				panic(fmt.Errorf("failed to decode %s: %v", ocr2Config.ConfigPublicKey, err))
+			}
+
+			configPkBytesFixed := [ed25519.PublicKeySize]byte{}
+			n = copy(configPkBytesFixed[:], configPkBytes)
+			if n != ed25519.PublicKeySize {
+				panic(fmt.Errorf("wrong num elements copied"))
+			}
+
+			onchainPkBytes, err := hex.DecodeString(strings.TrimPrefix(ocr2Config.OnchainPublicKey, "ocr2on_evm_"))
+			if err != nil {
+				panic(fmt.Errorf("failed to decode %s: %v", ocr2Config.OnchainPublicKey, err))
+			}
+
+			oracleIdentities[i] = ocr2config.OracleIdentityExtra{
+				OracleIdentity: ocr2config.OracleIdentity{
+					OnchainPublicKey:  onchainPkBytes,
+					OffchainPublicKey: offchainPkBytesFixed,
+					PeerID:            p2pKeys[0].PeerID,
+					TransmitAccount:   ocr2types.Account(ni.NodeAddress[0]),
+				},
+				ConfigEncryptionPublicKey: configPkBytesFixed,
+			}
+
+			ni.OcrSigningAddress = []string{common.HexToAddress(strings.TrimPrefix(ocr2Config.OnchainPublicKey, "ocr2on_evm_")).Hex()}
+			err = writeJSON(ni, strconv.Itoa(i)+".json")
+			if err != nil {
+				panic(fmt.Errorf("failed to write node info to JSON: %v", err))
+			}
+
+		}(i, cl)
+	}
+	wg.Wait()
+
+	// if node info is not in RDD and weiwatchers, don't proceed further
+	if !h.cfg.VerifyNodes {
+		return
+	}
+	nodeInfos := fetchNodeInfos(ctx, h.cfg.NodeConfigURL, log)
+	cnt := 0
+	for _, ni := range nodeInfos {
+		nodeAddr := ni.NodeAddress[0]
+		node := nodes[nodeAddr]
+		if node == nil {
+			continue
+		}
+		cnt++
+		diffs := 0
+
+		log.Infof("start comparing data for node %s", nodeAddr)
+		if node.CSAKeys[0].NodeAddress != ni.CSAKeys[0].NodeAddress {
```

Comment:
Does it make sense to implement a single `Equals` function on the `NodeInfo` type that just returns a boolean, and then move all of these comparisons into that function?

Reply:
done